moveit_planners/trajopt/include/problem_description.h

```diff
@@ -0,0 +1,286 @@
+#pragma once
+#include <tesseract_core/basic_env.h>
+#include <tesseract_core/basic_kin.h>
+#include <trajopt/common.hpp>
+#include <trajopt/json_marshal.hpp>
+#include <trajopt_sco/optimizers.hpp>
+#include <moveit/robot_model/robot_model.h>
+
+#include <moveit/planning_scene/planning_scene.h>
+
+namespace trajopt_interface
+{
+
+/**
+@brief Used to apply cost/constraint to joint-space velocity
+
+Term is applied to every step between first_step and last_step. It applies two limits, upper_limits/lower_limits,
+to the joint velocity subject to the following cases.
+
+* term_type = TT_COST
+** upper_limit = lower_limit = 0 - Cost is applied with a SQUARED error scaled for each joint by coeffs
+** upper_limit != lower_limit - 2 hinge costs are applied scaled for each joint by coeffs. If velocity < upper_limit and
+velocity > lower_limit, no penalty.
+
+* term_type = TT_CNT
+** upper_limit = lower_limit = 0 - Equality constraint is applied
+** upper_limit != lower_limit - 2 Inequality constraints are applied. These are both satisfied when velocity <
+upper_limit and velocity > lower_limit
+
+Note: coeffs, upper_limits, and lower_limits are optional. If a term is not given it will default to 0 for all joints.
+If one value is given, this will be broadcast to all joints.
+
+Note: Velocity is calculated numerically using forward finite difference
+
+\f{align*}{
+  cost = \sum_{t=0}^{T-2} \sum_j c_j (x_{t+1,j} - x_{t,j})^2
+\f}
+where j indexes over DOF, and \f$c_j\f$ are the coeffs.
+*/
+
+
+struct TermInfo;
+  //typedef std::shared_ptr<TermInfo> TermInfoPtr;
+MOVEIT_CLASS_FORWARD(TermInfo);
+
+class TrajOptProblem;
+typedef std::shared_ptr<TrajOptProblem> TrajOptProblemPtr;
+
+struct ProblemInfo;
+
+TrajOptProblemPtr ConstructProblem(const ProblemInfo&);
+
+
+enum TermType
+{
+  TT_COST = 0x1,      // 0000 0001
```

Comment:
please add comments explaining what these are

Reply:
They are explained in line 15 and below that